tests/std/tests/P0896R4_common_iterator_death/test.cpp

```
@@ -0,0 +1,100 @@
+
+    friend void iter_swap(const simple_input_iter&, const simple_input_iter&) {}
+};
+
+using CIT = common_iterator<simple_input_iter, default_sentinel_t>;
+
+void test_case_operator_dereference() {
+    CIT cit{default_sentinel};
+    (void) (*cit); // cannot dereference common_iterator that holds a sentinel
+}
+
+void test_case_operator_dereference_const() {
+    const CIT cit{default_sentinel};
+    (void) (*cit); // cannot dereference common_iterator that holds a sentinel
+}
+
+void test_case_operator_arrow() {
+    CIT cit{default_sentinel};
+    (void) (cit.operator->()); // cannot dereference common_iterator that holds a sentinel
+}
+
+void test_case_operator_preincrement() {
+    CIT cit{default_sentinel};
+    ++cit; // cannot pre increment common_iterator that holds a sentinel
+}
+
+void test_case_operator_increment() {
+    CIT cit{default_sentinel};
+    cit++; // cannot increment common_iterator that holds a sentinel
+}
+
+void test_case_iter_move() {
+    CIT cit{default_sentinel};
+    (void) ranges::iter_move(cit); // cannot iter_move common_iterator that holds a sentinel
+}
+
+void test_case_iter_swap_sentinel_left() {
+    CIT cit1{default_sentinel};
+    CIT cit2{};
+    (void) ranges::iter_swap(cit1, cit2); // cannot iter_swap common_iterators that hold a sentinel
+}
+
+void test_case_iter_swap_sentinel_right() {
+    CIT cit1{};
+    CIT cit2{default_sentinel};
+    (void) ranges::iter_swap(cit1, cit2); // cannot iter_swap common_iterators that hold a sentinel
+}
+
+int main(int argc, char* argv[]) {
+    std_testing::death_test_executive exec([] {});
+
+#if _ITERATOR_DEBUG_LEVEL != 0
+    exec.add_death_tests({
+        test_case_operator_dereference,
+        test_case_operator_dereference_const,
+        test_case_operator_arrow,
+        test_case_operator_preincrement,
+        test_case_operator_increment,
+        test_case_iter_move,
+        test_case_iter_swap_sentinel_left,
+        test_case_iter_swap_sentinel_right,
+    });

```

Missing death test coverage for `i == i` and `i - i` for an empty `common_iterator` `i`.

---

The problem is that I do not know how to create those tests cases.

To create a valueless_by_exception `common_iterator` I need to throw an exception. But those immediately terminate the death_test. So I am at loss how to test this here.

Not that the default constructed common_iterator holds a default constructed iterator

---

Created a tag type to construct a valueless `common_iterator`